go/consensus/tendermint/apps/roothash/genesis.go

```diff
@@ -30,19 +33,24 @@ func (app *rootHashApplication) InitChain(ctx *abci.Context, request types.Reque
 	// carved out it's entries by this point.
 
 	regState := registryState.NewMutableState(ctx.State())
-	runtimes, _ := regState.Runtimes()
+	runtimes, _ := regState.Runtimes(ctx)
 	for _, v := range runtimes {
 		ctx.Logger().Info("InitChain: allocating per-runtime state",
 			"runtime", v.ID,
 		)
-		app.onNewRuntime(ctx, v, &st)
+		if err := app.onNewRuntime(ctx, v, &st); err != nil {
+			return fmt.Errorf("failed to initialize runtime state: %w", err)
+		}
 	}
 
 	return nil
 }
 
 func (rq *rootHashQuerier) Genesis(ctx context.Context) (*roothashAPI.Genesis, error) {
-	runtimes := rq.state.Runtimes()
+	runtimes, err := rq.state.Runtimes(ctx)
+	if err != nil {
+		return nil, err
```

Comment:
in most places you've been wrapping these with context info

Reply:
Added.